hypervisor/arch/x86/lapic.c

```diff
@@ -45,20 +45,23 @@ static union lapic_base_msr saved_lapic_base_msr;
 
 static void clear_lapic_isr(void)
 {
-	uint32_t isr_reg = MSR_IA32_EXT_APIC_ISR0;
+	uint32_t acked;
+	uint32_t isr_reg;
 
 	/* This is a Intel recommended procedure and assures that the processor
 	 * does not get hung up due to already set "in-service" interrupts left
 	 * over from the boot loader environment. This actually occurs in real
 	 * life, therefore we will ensure all the in-service bits are clear.
 	 */
-	do {
-		if (msr_read(isr_reg) != 0U) {
-			msr_write(MSR_IA32_EXT_APIC_EOI, 0U);
-			continue;
+	for (isr_reg = MSR_IA32_EXT_APIC_ISR7; isr_reg >= MSR_IA32_EXT_APIC_ISR0; isr_reg--) {
+		for (acked = 0U; acked < 32U; acked++) {
+			if (msr_read(isr_reg) != 0U) {
+				msr_write(MSR_IA32_EXT_APIC_EOI, 0U);
+			} else {
```

Comment:
Here I prefer you add a variable  isr_val, and: isr_val =msr_read(isr_reg).
This way, you can remove break, and do condition check in for loop.